x-pack/plugins/log_explorer/public/components/dataset_selector/state_machine/state_machine.ts

```diff
@@ -39,56 +39,73 @@ export const createPureDatasetsSelectorStateMachine = (
               },
             },
             open: {
-              initial: 'listingIntegrations',
+              initial: 'integrationsTab',
               on: {
                 CLOSE: 'closed',
                 TOGGLE: 'closed',
+                SELECT_ALL_LOGS_DATASET: 'closed',
               },
               states: {
                 hist: {
                   type: 'history',
+                  history: 'deep',
                 },
-                listingIntegrations: {
-                  entry: ['storePanelId', 'retrieveSearchFromCache', 'maybeRestoreSearchResult'],
+                integrationsTab: {
+                  initial: 'listingIntegrations',
+                  entry: ['storeIntegrationsTabId'],
                   on: {
-                    CHANGE_PANEL: [
-                      {
-                        cond: 'isUnmanagedStreamsId',
-                        target: 'listingUnmanagedStreams',
-                      },
-                      {
-                        target: 'listingIntegrationStreams',
-                      },
-                    ],
-                    SCROLL_TO_INTEGRATIONS_BOTTOM: {
-                      actions: 'loadMoreIntegrations',
-                    },
-                    SEARCH_BY_NAME: {
-                      actions: ['storeSearch', 'searchIntegrations'],
-                    },
-                    SORT_BY_ORDER: {
-                      actions: ['storeSearch', 'sortIntegrations'],
-                    },
-                    SELECT_ALL_LOGS_DATASET: '#closed',
+                    SWITCH_TO_UNCATEGORIZED_TAB: 'uncategorizedTab',
                   },
-                },
-                listingIntegrationStreams: {
-                  entry: ['storePanelId', 'retrieveSearchFromCache', 'maybeRestoreSearchResult'],
-                  on: {
-                    CHANGE_PANEL: 'listingIntegrations',
-                    SEARCH_BY_NAME: {
-                      actions: ['storeSearch', 'searchIntegrationsStreams'],
+                  states: {
+                    hist: {
```

Comment:
**note:** an history state is required for this level too because the top-level `deep` history can't restore the nested states for the `integrationsTab` state. Having a shallow history at this level is the only way for the top-level history to restore the previous navigation level.